File: client/src/model/read/Quality.ts

export enum QualityType {
  Bool = 1,
  Int = 2,
  Enum = 3
}


// in the backend, this is QualityHeader, as full qualities do not transfer
// to the front end.
export interface Quality {
  name: string;
  type: QualityType;
  icon: string | null;
  description: string;
  hidden: boolean;
}

File: service/GoodNight.Service.Domain/Model/Read/Transfer/Quality.cs
using GoodNight.Service.Domain.Model.Expressions;

namespace GoodNight.Service.Domain.Model.Read.Transfer
{
  /// <summary>
  /// A QualityHeader is a short summary of a quality. It provides the
  /// most important information in a format suitable for serialisation.
  /// </summary>
  public record Quality(
    string Name,
    Type Type,
    string? Icon,
    string Description,
    bool Hidden);
}
